nautobot_ssot/integrations/vsphere/jobs.py

```diff
@@ -0,0 +1,179 @@
+#  pylint: disable=keyword-arg-before-vararg
+#  pylint: disable=too-few-public-methods
+#  pylint: disable=too-many-locals
+#  pylint: disable=abstract-method
+
+"""Job for vSphere integration with SSoT app."""
+
+from diffsync.enum import DiffSyncFlags
+from django.templatetags.static import static
+from django.urls import reverse
+from nautobot.apps.jobs import MultiObjectVar, ObjectVar
+from nautobot.extras.choices import (
+    SecretsGroupAccessTypeChoices,
+    SecretsGroupSecretTypeChoices,
+)
+from nautobot.extras.jobs import BooleanVar
+from nautobot.virtualization.models import Cluster
+
+from nautobot_ssot.integrations.vsphere.diffsync.adapters import (
+    NBAdapter,
+    VsphereDiffSync,
+)
+from nautobot_ssot.integrations.vsphere.models import SSOTvSphereConfig
+from nautobot_ssot.integrations.vsphere.utilities import VsphereClient, VsphereConfig
+from nautobot_ssot.jobs.base import DataMapping, DataSource
+
+name = "SSoT - Virtualization"  # pylint: disable=invalid-name
+
+
+def _get_vsphere_client_config(app_config, debug):
+    """Get Infoblox client config from the Infoblox config instance."""
+    username = app_config.vsphere_instance.secrets_group.get_secret_value(
+        access_type=SecretsGroupAccessTypeChoices.TYPE_REST,
+        secret_type=SecretsGroupSecretTypeChoices.TYPE_USERNAME,
+    )
+    password = app_config.vsphere_instance.secrets_group.get_secret_value(
+        access_type=SecretsGroupAccessTypeChoices.TYPE_REST,
+        secret_type=SecretsGroupSecretTypeChoices.TYPE_PASSWORD,
+    )
+    vsphere_client_config = VsphereConfig(
+        vsphere_uri=app_config.vsphere_instance.remote_url,
+        username=username,
+        password=password,
+        verify_ssl=app_config.vsphere_instance.verify_ssl,
+        vm_status_map=app_config.default_vm_status_map,
+        ip_status_map=app_config.default_ip_status_map,
+        vm_interface_map=app_config.default_vm_interface_map,
+        primary_ip_sort_by=app_config.primary_ip_sort_by,
+        ignore_link_local=app_config.default_ignore_link_local,
+        use_clusters=app_config.use_clusters,
+        sync_tagged_only=app_config.sync_tagged_only,
+        debug=debug,
+    )
+
+    return vsphere_client_config
+
+
+# pylint:disable=too-few-public-methods
+class VsphereDataSource(DataSource):  # pylint: disable=too-many-instance-attributes
+    """vSphere SSoT Data Source."""
+
+    debug = BooleanVar(description="Enable for more verbose debug logging")
+    config = ObjectVar(
+        model=SSOTvSphereConfig,
+        display_field="SSOT vSphere Config",
```

Comment:
Not sure why I'm not seeing "SSOT vSphere Config" as the field name. I'm seeing the default "Config" as the field name in the "Sync Now" menu. The image provided in [docs/images/vsphere_job.png](https://github.com/nautobot/nautobot-app-ssot/tree/ab_vsphere_integration/docs/images/vsphere_job.png) also says "Config" so I think this is unintended.